pkg/controller/daemon/daemon_controller.go

```diff
@@ -1239,7 +1239,7 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 	wantToRun, shouldSchedule, shouldContinueRunning = true, true, true
 	// If the daemon set specifies a node name, check that it matches with node.Name.
 	if !(ds.Spec.Template.Spec.NodeName == "" || ds.Spec.Template.Spec.NodeName == node.Name) {
-		return false, false, false, nil
+		return false, false, false, fmt.Errorf("specified node name does not match node.Name")
```

Comment:
How about "Spec.NodeName does not match node's name"?

Reply:
sounds good to me.